src/boot/efifirmware.c

```diff
@@ -0,0 +1,60 @@
+/* SPDX-License-Identifier: LGPL-2.1-or-later */
+
+#include "efifirmware.h"
+#include "util.h"
+#include <endian.h>
+
+static bool efifw_validate_header(const void *blob, size_t uki_efifw_len,
+                                  uint32_t *fwid_off, uint32_t *payload_off) {
+        if ((uintptr_t) blob % alignof(EfiFwHeader) != 0)
+                return false;
+        const EfiFwHeader *fw_header = ASSERT_PTR(blob);
+
+        if (le32toh(fw_header->base.magic) != FWHEADERMAGIC)
```

Comment:
i mean, if you insist on doing those LE conversions, then do them properly, i.e. change  the types of the fields in question from uint32_t to le32_t. (so that certain static analyzers can automatically detect this)

But frankly, this is efi stuff, efi is always by specification LE only, hence i simply wouldn't bother we endianess here, and simplify things.